test/elixir/test/test_helper.exs

```diff
@@ -259,6 +259,59 @@ defmodule CouchTestCase do
         opts = [pretty: true, width: 20, limit: :infinity, printable_limit: :infinity]
         inspect(resp, opts)
       end
+
+      def restart_cluster() do
+        resp = Couch.get("/_membership")
+        assert resp.status_code == 200
+        nodes = resp.body["all_nodes"]
+
+        nodes_ports =
+          Enum.reduce(nodes, [], fn node, acc ->
+            port = node_to_port(node)
+            [{node, port} | acc]
+          end)
+
+        tasks =
+          Enum.map(nodes_ports, fn {node, port} ->
+            Task.async(fn -> restart_node(node, port) end)
+          end)
+
+        Task.yield_many(tasks, length(nodes) * 5000)
+      end
+
+      def restart_node(node \\ "node1@127.0.0.1") do
+        port = node_to_port(node)
+        restart_node(node, port)
+      end
+
+      defp restart_node(node, port) do
+        url = "http://127.0.0.1:#{port}/_node/#{node}/_restart"
+        resp = Couch.post(url)
+        assert HTTPotion.Response.success?(resp)
+        assert resp.body["ok"]
+        # make sure node went down. we assuming the node can't bounce quick
```

Comment:
If we want to be extra cautious we could use `uptime` to detect node restart.
```
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
218
$ http --auth=adm:pass POST :15984/_node/node2@127.0.0.1/_restart
HTTP/1.1 200 OK
Cache-Control: must-revalidate
Content-Length: 12
Content-Type: application/json
Date: Fri, 07 Dec 2018 14:59:18 GMT
Server: CouchDB/2.3.0-fcd62ed-dirty (Erlang OTP/19)
X-Couch-Request-ID: b5b6d5b302
X-CouchDB-Body-Time: 0

{
    "ok": true
}

$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
null
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
null
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
null
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
null
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
0
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
1
$ http --auth=adm:pass :15984/_node/node2@127.0.0.1/_system | jq .uptime
2
```

Reply:
I think this is an overkill. Checking that node went down, then got back should be enough to make sure we have our restart.